es-notebooks/src/test/kotlin/com/amazon/opendistroforelasticsearch/notebooks/PluginRestTestCase.kt

```diff
@@ -0,0 +1,232 @@
+    }
+
+    @Throws(IOException::class)
+    protected open fun configureHttpsClient(builder: RestClientBuilder, settings: Settings) {
+        val headers = ThreadContext.buildDefaultHeaders(settings)
+        val defaultHeaders = arrayOfNulls<Header>(headers.size)
+        var i = 0
+        for ((key, value) in headers) {
+            defaultHeaders[i++] = BasicHeader(key, value)
+        }
+        builder.setDefaultHeaders(defaultHeaders)
+        builder.setHttpClientConfigCallback { httpClientBuilder: HttpAsyncClientBuilder ->
+            val userName = System.getProperty("user")
+            val password = System.getProperty("password")
+            val credentialsProvider: CredentialsProvider = BasicCredentialsProvider()
+            credentialsProvider.setCredentials(AuthScope.ANY, UsernamePasswordCredentials(userName, password))
+            try {
+                return@setHttpClientConfigCallback httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider)
+                    // disable the certificate since our testing cluster just uses the default security configuration
+                    .setSSLHostnameVerifier(NoopHostnameVerifier.INSTANCE)
+                    .setSSLContext(
+                        SSLContextBuilder.create()
+                            .loadTrustMaterial(null) { _: Array<X509Certificate?>?, _: String? -> true }
+                            .build())
+            } catch (e: Exception) {
+                throw RuntimeException(e)
+            }
+        }
+        val socketTimeoutString = settings[CLIENT_SOCKET_TIMEOUT]
+        val socketTimeout = TimeValue.parseTimeValue(socketTimeoutString ?: "60s", CLIENT_SOCKET_TIMEOUT)
+        builder.setRequestConfigCallback { conf: RequestConfig.Builder ->
+            conf.setSocketTimeout(
+                Math.toIntExact(
+                    socketTimeout.millis
+                )
+            )
+        }
+        if (settings.hasValue(CLIENT_PATH_PREFIX)) {
+            builder.setPathPrefix(settings[CLIENT_PATH_PREFIX])
+        }
+    }
+
+    @Before
+    @Throws(InterruptedException::class)
+    open fun setup() {
+        if (client() == null) {
+            initClient()
+        }
+    }
+
+    fun executeRequest(
+        method: String,
+        url: String,
+        jsonString: String,
+        expectedRestStatus: Int? = null
+    ): JsonObject {
+        val request = Request(method, url)
+        request.setJsonEntity(jsonString)
+        val restOptionsBuilder = RequestOptions.DEFAULT.toBuilder()
+        restOptionsBuilder.addHeader("Content-Type", "application/json")
+        request.setOptions(restOptionsBuilder)
+        return executeRequest(request, expectedRestStatus)
+    }
+
+    private fun executeRequest(request: Request, expectedRestStatus: Int? = null): JsonObject {
+        val response = try {
+            client().performRequest(request)
+        } catch (exception: ResponseException) {
+            exception.response
```

Comment:
If this exception is hit, will a value be returned? I don't see a return statement here

Reply:
not very sure about kotlin syntax. I think if exception is caught and not thrown upwards, the code will keep running after catch block